crates/apollo-parser/src/error.rs

```diff
@@ -34,10 +34,37 @@ use std::fmt;
 /// [annotate_snippets]: https://github.com/apollographql/apollo-rs/blob/a7f616454a53dcb8496725ceac6c63eacddefb2c/crates/apollo-parser/examples/annotate_snippet.rs
 /// [miette crate]: https://docs.rs/miette/3.2.0/miette/index.html
 
-#[derive(PartialEq, Eq, Clone)]
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub(crate) enum ErrorData {
+    Eof,
+    LimitExceeded,
+    Text(String),
+}
+
+impl ErrorData {
+    pub fn len(&self) -> usize {
+        match self {
+            Self::Eof | Self::LimitExceeded => 0,
+            Self::Text(text) => text.len(),
+        }
+    }
+}
+
+impl fmt::Display for ErrorData {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Eof => write!(f, "EOF"),
+            Self::LimitExceeded => Ok(()),
```

Comment:
What does `Ok(())` mean in the context of displaying limit errors?

Reply:
doesn't display any data for the error. so it results in something like `ERROR@245460:245460 "token limit reached, aborting parsing" `.